dts/arm/nxp/nxp_mcxw71.dtsi

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <arm/armv8-m.dtsi>
+#include <zephyr/dt-bindings/clock/scg_k4.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv8m-mpu";
+				reg = <0xe000ed90 0x40>;
+			};
+		};
+	};
+
+	soc {
+		ctcm: sram@14000000 {
+			ranges = <0x0 0x14000000 DT_SIZE_K(16)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			ctcm0: code_memory@0 {
+				compatible = "mmio-sram";
+				reg = <0x0 DT_SIZE_K(16)>;
+			};
+		};
+
+		stcm: sram@30000000 {
+			ranges = <0x0 0x30000000 DT_SIZE_K(112)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			stcm0: system_memory@0 {
+				compatible = "mmio-sram";
+				reg = <0x0 DT_SIZE_K(64)>;
+			};
+
+			stcm1: system_memory@1a000 {
+				compatible = "zephyr,memory-region","mmio-sram";
+				reg = <0x1a000 DT_SIZE_K(8)>;
+				zephyr,memory-region = "RetainedMem";
+			};
+		};
+
+		smu2: sram@489C0000 {
+			ranges = <0x0 0x489C0000 DT_SIZE_K(40)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+		};
+
+		peripheral: peripheral@50000000 {
+			ranges = <0x0 0x50000000 DT_SIZE_K(512)>;
+
+			fmu: flash-module@20000 {
+				ranges = <0x0 0x10000000 DT_SIZE_M(1)>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				compatible = "nxp,fmu-k4";
+				reg = <0x20000 0x1000>;
+
+				flash: flash@0 {
+					reg = <0x0 DT_SIZE_M(1)>;
+					compatible = "soc-nv-flash";
+					write-block-size = <16>;
+					erase-block-size = <8192>;
+				};
+			};
+		};
+
+		fast_peripheral: peripheral@58000000 {
```

Comment:
I'd prefer we group these, unless there's a strong reason not to. That will make the nonsecure definition of this part a little simpler, and is consistent with our other M33 SOCs.

Reply:
ok, I was thinking the cortex-m3 DTS would then be able to delete the &fast_peripheral bus node to have an accurate DT since it can't access that bridge but it looks like in the RM it says even though the radio subsystem is documented pretty well and is very flexible, that the firmware programmed to the radio NBU is intended to be developed by NXP ? so maybe we don't want to support the M3 target for this chip on zephyr anyways, so we can pretend that there is not two peripheral bridges for the DT if it's that important

Reply:
Ah, that is quite a good reason. I'm fine to keep it as is then

Reply:
i'll just change it for now because I don't think we will have the m3 target any time soon, if ever

Reply:
FYI, On W71, the m3 target is not supposed to be open, so we shouldn't consider it as a target for zephyr.
It will be different on W72, where the NBU will run on a CM33+nodsp+nofpu and will be open.